src/pages/Home/Home.tsx

```diff
@@ -78,15 +77,23 @@ const Home = () => {
       perPage: shelters.perPage,
       search: qs.stringify(filterData),
     };
-
-    refresh(
-      {
-        params: params,
-      },
-      true
-    );
+    refresh({ params }, true);
   }, [refresh, filterData, shelters.filters, shelters.page, shelters.perPage]);
 
+  useEffect(() => {
+    const storedFilterData = loadFilterData();
+    const Keys = ['search', 'priority', 'cities']; 
```

Comment:
Como essa variável keys não sofre alterações, ela pode ser definida fora do render, não sendo recriada a todo momento.

Tenho a impressão que essa storedFilterData (que não é apenas dados armazenados) pode ser criada no início do componente e seu valor passado para o useState e reutilizada depois aqui.

Reply:
Vou realizar esses ajuste, válido demais seu comentário